src/components/TransactionImport/Form.tsx

```diff
@@ -0,0 +1,120 @@
+import { useTranslation } from 'react-i18next'
+import { useState } from 'react'
+import { Link } from 'react-router-dom'
+import { Account, TransactionPreview, Translation } from '../../types'
+import { parseFile } from '@envelope-zero/ynap-parsers'
+import { checkStatus, parseJSON } from '../../lib/fetch-helper'
+import Error from '../Error'
+import FormFields from '../FormFields'
+import FormField from '../FormField'
+import Autocomplete from '../Autocomplete'
+import LoadingSpinner from '../LoadingSpinner'
+import { safeName } from '../../lib/name-helper'
+
+type Props = {
+  accounts: Account[]
+  isLoading: boolean
+  setIsLoading: (isLoading: boolean) => void
+  setResult: (result: TransactionPreview[], targetAccountId: string) => void
+}
+
+const Form = ({ accounts, isLoading, setIsLoading, setResult }: Props) => {
+  const { t }: Translation = useTranslation()
+  const [error, setError] = useState('')
+  const [accountId, setAccountId] = useState('')
+
+  return (
+    <form
+      encType="multipart/form-data"
+      onSubmit={event => {
+        event.preventDefault()
+        const file = (event.target as any).file.files[0] // TODO: without `any`
+
+        parseFile(file)
+          .then(result => {
+            // Get the data and transform it to FormData for the backend
+            const data = new FormData()
+            data.append(
+              'file',
+              new File([new Blob([result[0].data])], 'ynap-transactions.csv', {
+                type: 'text/csv;charset=utf-8;',
+              })
+            )
+
+            fetch(`/api/v1/import/ynab-import-preview?accountId=${accountId}`, {
+              method: 'POST',
+              body: data,
+            })
+              .then(checkStatus)
+              .then(parseJSON)
+              .then(body => {
+                setIsLoading(false)
+                if (error) {
+                  setError('')
+                }
+                setResult(body.data, accountId)
+              })
+              .catch(error => {
+                setIsLoading(false)
+                setError(error.message)
+              })
+          })
+          // TODO: If the file is already in the YNAB Import format, we can pass it on directly
```

Comment:
If we can't parse the file, we should try to detect if it is already parsed and in that case, send it to the backend directly